### PR TITLE
#33 Added Death Functionality

### PR DESCRIPTION
**Pull request by VeikkaH**
Added death functionality, death screen and tests.

### CODE REVIEW

File: Assets/Scripts/DeathScreen.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    void Update()
+    {
+        /* if(Input.GetKeyDown(KeyCode.Escape))
+        {
+            if(!panel.activeSelf)
+            {
+                ShowDeathScreen();
+            }
+        } */
```
Comment (Usmo) [inline Assets/Scripts/DeathScreen.cs:23]: This commented section is not needed

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -6,15 +6,24 @@ public class Player : MonoBehaviour
...
     public int health = 1;
 
+    public bool isAlive = true;
+
     float horizontalMove = 0f;
     bool jump = false;
-
+    void Start()
+    {
+        transform.position = startPosition.transform.position;
```
Comment (Usmo) [inline Assets/Scripts/Player.cs:25]: Maybe null check here so tests made by others don't need any modifications. For example:
if(startPosition != null) {
    transform.position = startPosition.transform.position;
}

File: Assets/Tests/PlayMode/CharacterMovement.cs [hunk 3]
```diff
@@ -10,11 +10,13 @@ public class CharacterMovement
     public IEnumerator positive_movement_moves_right()
     {
         // ARRANGE
+        GameObject startPosition = new GameObject("Start");
```
Comment (Usmo) [inline Assets/Tests/PlayMode/CharacterMovement.cs:13]: This test file dont need any changes, if player class has null checks implemented as suggested above. :D

File: Assets/Scripts/Player.cs [hunk 4]
```diff
@@ -33,4 +43,13 @@ void FixedUpdate()
         jump = false;
     }
 
+    public void takeDamage()
+    {
+        health--;
+        if (health <= 0)
+        {
+            isAlive = false;
+            deathScreen.ShowDeathScreen();
```
Comment (Usmo) [inline Assets/Scripts/Player.cs:52]: Null check also here to check if deathScreen is null.

File: Assets/Tests/PlayMode/EnemyTests.cs [hunk 5]
```diff
@@ -14,11 +14,19 @@ public IEnumerator EnemyCollisionDamagesPlayer()
         GameObject playerObject = new GameObject("playerObject", typeof(Rigidbody2D), typeof(BoxCollider2D), typeof(Player), typeof(CharacterController));
         playerObject.tag = "Player";
         CharacterController controller = playerObject.GetComponent(typeof(CharacterController)) as CharacterController;
+
```
Comment (Usmo) [inline Assets/Tests/PlayMode/EnemyTests.cs:17]: This test file dont need any changes, if player class has null checks implemented as suggested above. :D

**Review by Usmo — CHANGES_REQUESTED** [review]
SampleScene merge problem. Needs to revert all changes to SampleScene -> then rebase to main branch -> then redo deathscreen things to SampleScene

**Comment by VeikkaH** [comment]
Everything should be fixed, working well, comments noted

**Review by Usmo — APPROVED** [review]

**Pull request merged**